cilium/CFP-33716-distributed-packet-tracing.md

```diff
@@ -0,0 +1,586 @@
+# CFP-2024-06-12: Packet Tracing
+
+## Meta
+**SIG:**  
+**Sharing:** Public  
+**Begin Design Discussion:** Jun 12, 2024  
+**End Design Discussion:**  
+**Cilium Release:**  
+**Authors:** Ben Bigdelle <bigdelle@google.com>, Mark St. John <markstjohn@google.com> based on the work of Nathan Perkins <nathanperkins@google.com>  
+**Reviewers:**  
+
+## Summary
+We aim to enhance cross-cluster and multi-system observability for flows that contain an `ip_trace_id` IP Option (referred to as a tagged flow), thus supporting tracing across networks. These tagged flows will trigger adding the ID value to Flow events messages and client (Hubble / Cilium Monitor) representation. This allows correlating flow events across systems which integrate with the span observability.
```

Comment:
If this is a standardized IP option, please add a link to the relevant RFC where the option is defined.

Reply:
We mean to say that a user defines which IP option is being used to store their `ip_trace_id`. There isn't a dedicated `ip_trace_id` field. A user can configure which IP option type is most appropriate to store a series of bytes into as the `ip_trace_id`, which we parse from accordingly. Any well-defined IP Option field type can be used (https://www.iana.org/assignments/ip-parameters/ip-parameters.xhtml)